app/src/main/java/com/muzima/model/AvailableForm.java

```diff
@@ -42,4 +43,17 @@ public boolean isRegistrationForm(){
         }
         return false;
     }
+    public boolean hasDescriminator(String discriminator){
```

Comment:
I think you could simply compare supplied discriminator against getDiscriminator().